### ISSUE TITLE
Traverse pagination of issue results if > 100 results

### ISSUE THREAD

**Issue by miketaylr** [past/tsci#8]

**Comment by past** [comment]
I'm getting this at master now:
```
Fetched bug data for website google.com
Fetched bug data for website youtube.com
(node:12647) UnhandledPromiseRejectionWarning: HttpError: Only the first 1000 search results are available
    at /Users/past/src/tsci/node_modules/@octokit/request/dist-node/index.js:85:23
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
    at async Job.doExecute (/Users/past/src/tsci/node_modules/bottleneck/light.js:387:18)
(node:12647) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:12647) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```

**Comment by miketaylr** [comment]
> Only the first 1000 search results are available

Interesting. c.f., https://developer.github.com/v3/search/#about-the-search-api

There's no way there's more than 1000 issues for youtube, so possibly we're trying to paginate on *all* issues?

**Comment by miketaylr** [comment]
https://github.com/past/tsci/blob/222aeb4153d93aacfdf23216c432d817ed91589b/bugs.js#L134-L152

I'm wondering if we actually need to do pagination traversal if we're just after the result total. My bad if not... 🙈. Will investigate.

**Comment by past** [comment]
This is the failing request:
`https://api.github.com/search/issues?per_page=100&q=is%3Aissue+milestone%3Aduplicate+repo%3Awebcompat%2Fweb-bugs%2F&per_page=100&page=11`
So page 11 would be results 1000-1100.

**Comment by past** [comment]
Found with:
```
diff --git a/bugs.js b/bugs.js
index 27ec5c4..6b92d37 100644
--- a/bugs.js
+++ b/bugs.js
@@ -140,7 +140,12 @@ async function getAllGitHubResultsFor(query, params = {}) {
     params.page = 0; // Note: this parameter is 1-based, not 0-based
     while (expected === undefined || results.length < expected) {
         ++params.page;
-        const response = await query.call(this, params);
+        let response;
+        try {
+            response = await query.call(this, params);
+        } catch (err) {
+            console.log(util.inspect(err, { showHidden: false, depth: null }));
+        }
         if (!response.data) {
             console.log(util.inspect(response, { showHidden: false, depth: null }))
             throw new Error("GitHub query failed!");

```
We should probably add the try/catch even after we fix this.

**Comment by miketaylr** [comment]
> I'm wondering if we actually need to do pagination traversal if we're just after the result total.

I think the answer is, it depends. Or maybe not at all?

For `getWebcompat`, we're just after `response.data.total_count`, so we can save time/requests by not fetching pages. 

@wisniewskit can you tell me if my understanding of `getDuplicates` is correct here?

```
1. grab all the bugzilla bugs that have a see also w/ webcompat.com or github.com/webcompat in "See Also"
2. make sure those bugs are within the min/max date range 
3. add those bugs to a `githubCandidates` array (as a bugzilla id, webcompat id tuple)
4. do a bunch of searches against the github search API for the webcompat id and see if their milestone is a duplicate, if so, toss them in the `dupedBzIds` set
```

So maybe we don't need pagination traversal here either?

>  `https://api.github.com/search/issues?per_page=100&q=is%3Aissue+milestone%3Aduplicate+repo%3Awebcompat%2Fweb-bugs%2F&per_page=100&page=11`

I wonder what this request is all about. We don't want to look at all duplicate issues, just all issues for a given website, right? It seems like the search should be something like `https://api.github.com/search/issues?q=31802+31799+is%3Aissue+milestone%3Aduplicate+repo%3Awebcompat%2Fweb-bugs&type=Issues` (that will return 0 results, because those issues aren't dupes, just an example)

**Comment by miketaylr** [comment]
Doing some naive logging,

```
https://api.github.com/search/issues?per_page=100&q=is%3Aissue+milestone%3Aduplicate+repo%3Awebcompat%2Fweb-bugs%2F+5576+4064+8436+8485+8991+7290+8877+9849+9521+10533+10449+12326+12368+12353+12420+13013+13227+13325+13592+13591+13595+13718+14033+14295+14459+14559+14696+10398+14809+14964+14999+14981+14984
https://api.github.com/search/issues?per_page=100&q=is%3Aissue+milestone%3Aduplicate+repo%3Awebcompat%2Fweb-bugs%2F+15154+15930+16212+16517+16668+16760+17265+17385+17719+18804+19462+19558+19682+20277+21244+23675+23359+23764+23967+23980+24508+24626+24653+26737+27376+28344+29676+1347+3137+8012+15071+12439
https://api.github.com/search/issues?per_page=100&q=is%3Aissue+milestone%3Aduplicate+repo%3Awebcompat%2Fweb-bugs%2F+15676+15705+15710+15847+15863+15862+15837+15823+15792+15677+15644+15643+15950+15904+15979+15900+16029+16024+16087+16115+16132+16122+16170+16211+16298+16310+16568+16902+18284+18285+18270+18268
https://api.github.com/search/issues?per_page=100&q=is%3Aissue+milestone%3Aduplicate+repo%3Awebcompat%2Fweb-bugs%2F+18265+18264+18262+18260+18259+18256+18251+18244+18249+18267+18300+18296+18295+18293+17007+17008+17009+17019+17067+20551+29656+21295+17019+26945+28357
Fetched bug data for website google.com
https://api.github.com/search/issues?per_page=100&q=is%3Aissue+milestone%3Aduplicate+repo%3Awebcompat%2Fweb-bugs%2F+17549+18885
Fetched bug data for website youtube.com
https://api.github.com/search/issues?per_page=100&q=is%3Aissue+milestone%3Aduplicate+repo%3Awebcompat%2Fweb-bugs%2F
```

It looks like YouTube is returning no results, so we're just searching for all of them (unintentionally). I'm confused about the YT result though, this quick search shows at least 2 issues with a webcompat see also.

https://bugzilla.mozilla.org/buglist.cgi?f1=see_also&bug_file_loc_type=allwordssubstr&list_id=14731439&o1=anywords&resolution=---&classification=Client%20Software&classification=Developer%20Infrastructure&classification=Components&classification=Server%20Software&classification=Other&bug_file_loc=youtube.com&query_format=advanced&v1=webcompat

**Comment by wisniewskit** [comment]
@miketaylr yes, that's what the query should be doing. I'll take a look to see why it's not sending the right query.

**Comment by wisniewskit** [comment]
Based on the symptoms (it not actually having any IDs, I'm guessing that it's just hitting a case where there *are* no IDs to search for (no dupes). I'll try to find the case where that's happening.

**Comment by wisniewskit** [comment]
Yep, that's it. Simple logic error. When it hits Netflix, it bombs, because it ends up adding a search-query that has no IDs. This diff fixes that:
```
diff --git a/bugs.js b/bugs.js
index 71f7fca..2cee04b 100644
--- a/bugs.js
+++ b/bugs.js
@@ -265,7 +265,9 @@ const getDuplicates = async (website, bugzillaKey, githubKey, minDate, maxDate)
         searchQuery += "+" + githubId;
         searchMapGhToBz.set(parseInt(githubId), bzId);
     }
-    searches.push([searchQuery, searchMapGhToBz]);
+    if (searchQuery !== baseSearchQuery) {
+      searches.push([searchQuery, searchMapGhToBz]);
+    }
 
     const dupedBzIds = new Set();
     const octokit = getOctokitInstance(githubKey);
```

**Comment by wisniewskit** [comment]
I've included the fix in this as part of #20.

**Comment by wisniewskit** [comment]
This should be fixed, now that #20 has landed.
